client/my-sites/media-library/media-image.tsx

```diff
@@ -1,16 +1,15 @@
-import * as React from 'react';
 import ImagePreloader from 'calypso/components/image-preloader';
 import MediaFile, { MediaFileProps } from './media-file';
 import { RenderedComponent } from './proxied-image';
 
-const MediaImage: React.FC< MediaFileProps > = function MediaImage( props: MediaFileProps ) {
-	let component: RenderedComponent = props.component;
+function MediaImage( props: MediaFileProps ) {
+	const component: RenderedComponent = props.component;
 	if ( component === 'img' && props.placeholder ) {
-		component = ImagePreloader;
+		return <MediaFile { ...props } component={ ImagePreloader } />;
```

Comment:
This early return should be the same behavior as the old reassignment, except that it satisfies TypeScript because the `component` variable doesn't have to be used to hold two different types.